### PR TITLE
Database lesson

### PR DESCRIPTION
**Pull request by AydinHassan**
TODO
- [x] - Rebase
- [x] - Do something a bit more challenging with output, (concat names, age groups, select all over 20)
- [x] - Ask to update some values, use `SelfCheck` to verify data was updated.
- [x] - Alter query binding to use `execute` parameters
- [x] - Unit Tests
- [x] - Problem
- [x] - Create Trait for common lesson tasks eg `getTemporaryPath()`
- [x] - Create Abstract lesson which implements `getSolution()` & `getProblem()` based on class name. Implement empty `tearDown()` method

### CODE REVIEW

**Comment by AydinHassan** [comment]
Requires https://github.com/php-school/php-workshop/pull/45

**Comment by AydinHassan** [comment]
@mikeymike Ready for review, will need to switch php-workshop dependency back, one other PR is merged.

File: res/problems/database-read/problem.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+Write a program that receives a database connection string (DSN). Connect to the database, query it and update some data.
+
+Display the information of all the users in the database table `users` who's is are over 30. Print out each row as a new line formatted like:
+
+`User: Jim Morrison Age: 27 Sex: Male` followed by a line break.
+
+Finally you will be given a random name as the second argument to your program, you should update the row in the `users` table 
+which corresponds to this name. You should change the name to `David Attenborough`.
```
Comment (mikeymike) [inline res/problems/database-read/problem.md]: are these newline chars?  L14/39 too. Better to remove and let the terminal wrap.

File: app/bootstrap.php [hunk 2]
```diff
@@ -48,7 +49,7 @@
         |_''_|
 
       PHP SCHOOL
-LEARNING FOR ELEPHPANTS
+LEARNING FOR ELEPHANTS
```
Comment (mikeymike) [inline app/bootstrap.php]: fyi this was intentional :laughing:

**Comment by mikeymike** [comment]
@AydinHassan Really nice man, works really well using `DatabaseExerciseCheck` :heart:

**Pull request merged**